lib/components/Input/base/index.tsx

```diff
@@ -0,0 +1,149 @@
+    inputSize?: TMediumSizes;
+    status?: "neutral" | "success" | "error";
+    disabled?: boolean;
+    variant?: "fill" | "outline";
+    leftStatusMessage?: string;
+    rightStatusMessage?: string;
+    textAlignment?: "left" | "center";
+    label?: string;
+    value?: string;
+}
+
+const IconSize = {
+    sm: { width: 24, height: 24 },
+    md: { width: 24, height: 24 },
+};
+
+const statusIconColors = {
+    neutral: "status-icon--neutral",
+    success: "status-icon--success",
+    error: "status-icon--error",
+};
+
+const Input = forwardRef<HTMLInputElement, InputProps>(
+    (
+        {
+            type = "text",
+            inputSize = "md",
+            className,
+            status = "neutral",
+            disabled = false,
+            variant = "outline",
+            icon: Icon,
+            leftStatusMessage,
+            rightStatusMessage,
+            textAlignment = "left",
+            label,
+            statusIcon: StatusIcon,
+            onChange,
+            ...rest
+        },
+        ref,
+    ) => {
+        const [hasValue, setHasValue] = useState(false);
+
+        return (
+            <div className="quill-input__container">
+                <div
+                    data-has-value={hasValue}
+                    className={clsx(
+                        className,
+                        `quill-input__wrapper`,
+                        `quill-input__wrapper__variant--${variant}`,
+                        variant === "fill" && `status--${status}`,
+                        `quill-input__wrapper__size--${inputSize}`,
+                        `quill-input__wrapper__status--${status}`,
+                    )}
+                >
+                    {Icon && (
+                        <div>
+                            <Icon {...IconSize[inputSize]} />
+                        </div>
+                    )}
+                    <input
+                        {...rest}
+                        type={type}
+                        className={clsx(
+                            "input",
+                            "peer",
+                            `input__align--${textAlignment}`,
+                            `input__size--${inputSize}`,
+                        )}
+                        disabled={!!disabled}
+                        onChange={(e) => {
+                            setHasValue(!!e.target.value);
+                            onChange?.(e);
+                        }}
+                        ref={ref}
+                    />
+                    {label && inputSize === "md" && (
+                        <label
+                            className={clsx(
+                                "label",
+                                `label__status--${status}`,
+                            )}
+                        >
+                            {label}
+                        </label>
+                    )}
+                    {StatusIcon && (
+                        <div>
```

Comment:
Do we need this `<div>`?
Ignore if not relevant

Reply:
@Nuzhy-Deriv  i guess `className` should be assigned to the `div` instead of the icon itself in order for the icon to have a correct size and alignment. Otherwise, if it works perfectly as is, `div` looks unnecessary :)